drivers/srf02/srf02.c

```diff
@@ -86,22 +78,24 @@ int srf02_init(srf02_t *dev, i2c_t i2c, uint8_t addr)
     return 0;
 }
 
-void srf02_trigger(const srf02_t *dev, srf02_mode_t mode)
+int srf02_trigger(const srf02_t *dev, srf02_mode_t mode)
 {
+    int status;
     /* trigger a new measurement by writing the mode to the CMD register */
     DEBUG("[srf02] trigger new reading\n");
     i2c_acquire(dev->i2c);
-    i2c_write_reg(dev->i2c, dev->addr, REG_CMD, mode);
+    status = i2c_write_reg(dev->i2c, dev->addr, REG_CMD, mode, 0);
     i2c_release(dev->i2c);
+    return status;
 }
 
 uint16_t srf02_read(const srf02_t *dev)
 {
-    uint8_t res[2];
+    uint8_t res[2] = {0xFF, 0xFF};
```

Comment:
why `0xff` instead of `0x0`, just curious 😄

Reply:
ah, you use that below to get invalid/failed reads, but then see my other comment on how this (IMHO) should be fixed instead.